src/ToolBox/SOS/Strike/disasm.cpp

```diff
@@ -530,7 +530,7 @@ INT_PTR ParseHexNumber (__in_z char *ptr, ___out char **endptr)
         endptr1 = endptr2;
     }
     // if the hex number was specified as 000006fbf9b70f50, an overflow occurred
-    else if (ULONG_MAX == value1 && errno == ERANGE)
+    else if (ULONG_MAX == (ULONG)value1 && errno == ERANGE)
```

Comment:
You could use "ULONG_PTR_MAX" instead of the cast.

Reply:
@mikem8361 ULONG_MAX is always 32 bit while ULONG_PTR_MAX is 64 bit / 32 bit based on the target bitness. So it seems it would not work as expected.

Reply:
does this (https://github.com/dotnet/coreclr/compare/09181b3dd4b8cd1e2fea7a6ca2b6234b10578d65..a8b3565d5a2626e3ffb7034b5d3a95b050eb5a6a) work?

Reply:
Issue here is signedness BTW. We'll still need a cast. Which cast is the question?